packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4369,67 +4369,343 @@ class PositionedDirectional extends StatelessWidget {
+///     might have been set to._ Any size you specify for the main axis
+///     of a widget that you then put in a [Flexible] with it's fit set to
+///     [FlexFit.tight] will be ignored, no matter if that size is larger or
+///     smaller than the leftover space. The child will always be as large
+///     as the space that was reserved for its parent [Expanded].
+///
+///    The only  children who should have a tight fit are the [Expanded]s.
+///    It is considered bad to use a [Flexible] and set its fit to tight
+///    because doing so creates unnecessary confusion, and this could be
+///    a reason for having one's coffee privileges revoked. If you need a
+///    tight fit, use an [Expanded] and other developers will understand
+///    what is happening at a glance.
+///
+///    Children with a fit that is [FlexFit.loose] are given loose constraints.
+///    The min size is not as big as the max, and the child is allowed to be
+///     the size it wants to be (in the main axis). The only children this
+///    should apply to are [Flexible]s.
+///
+///   An edge case was mentioned above, regarding using a [Flexible]
+///     and an [Expanded] in the same [Flex]. What happens is the
+///     [Expanded] will use all of the space reserved for it in Step 2, but
+///     the [Flexible] might not use all of the space reserved for it. Oftentimes,
+///     a person will use an [Expanded] thinking it will use _all_ of the
+///     leftover space in a [Flex], but this situation will not do that. I.E.:
+///
+///     - A [Column] is in a parent that is 300 high, with tight constraints.
+///     - One child is a box with a set height of 100.
+///     - Leftover space in the [Column] is 200.
+///     - There is one [Expanded] with a [flex] of 1.
+///     - There is one [Flexible] with a [flex] of 1.
+///     - The flexes are equal, so half of the space (100) is reserved for each.
+///     - Each has a child that is 50 high.
+///     - The [Expanded] will ignore the child's height and stretch to 100 high.
+///     - The [Flexible] will not stretch its child, and will be 50 high.
+///
+///     When you look at the result that is rendered, it seems the [Column]
+///     is short by 50, because all the children you can see add up to only
+///     250. However, the [Column] is not short, it is the full 300 high. What
+///     is happening is the [Column] is 300 high but all of its children only
+///     add up to 250 high. The [Expanded] _did_ use all of the space it
+///     was offered, but it was only offered 100.
+///
+///     The reason the [Column] looks short is that it has a transparent
+///     background, and you're looking _through_ the [Column] in the
+///     section that wasn't used. This makes it appear the [Column]
+///     is only 250 high, but it's actually 300, with 50 of that being
+///     transparent.
+///
+///     That is the edge case that can be confusing, if you're expecting
```

Comment:
```suggestion
///     That is one edge case that can be confusing, if you're expecting
```
There are undoubtedly more. :-)